### PR TITLE
Feat/vadc sprint21

### PR DESCRIPTION
**Pull request by jarvisraymond-uchicago**
Jira Ticket: [VADC-216](https://ctds-planx.atlassian.net/browse/VADC-216)
Jira Ticket: [VADC-267](https://ctds-planx.atlassian.net/browse/VADC-267)
Jira Ticket: [VADC-243](https://ctds-planx.atlassian.net/browse/VADC-243)
Jira Ticket: [VADC-83](https://ctds-planx.atlassian.net/browse/VADC-83)

### New Features
This implements the features associated with tickets for VADC Sprint 21 Frontend Work: 

* gwas V2 workflow progress bar 
* Add new command that only runs eslint on edited files in current git branch
* Add Attrition Table below Steps (basic setup & styling only)
* UI feature - Euler diagram for evaluating cohort selections in dichotomous covariate or dichotomous outcome steps

### Dependency updates
- Added "@upsetjs/venn.js": "^1.4.2"
-  Requires dependencies overrides: 
      "uglify-js": "3.14.3",
      "ansi-regex": "3.0.1"

### CODE REVIEW

File: src/Analysis/GWASV2/Shared/CohortsOverlapDiagram/CohortsOverlapDiagram.stories.jsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  // msw mocking:
+  msw: {
+    handlers: [
+      rest.post('http://:cohortmiddlewarepath/cohort-middleware/cohort-stats/check-overlap/by-source-id/:sourceid/by-cohort-definition-ids/:cohortdefinitionA/:cohortdefinitionB', (req, res, ctx) => {
+        const { cohortmiddlewarepath } = req.params;
+        const { cohortdefinitionA } = req.params;
+        const { cohortdefinitionB } = req.params;
+        console.log(cohortmiddlewarepath);
+        console.log(cohortdefinitionA);
+        console.log(cohortdefinitionB);
```
Comment (mfshao) [inline src/Analysis/GWASV2/Shared/CohortsOverlapDiagram/CohortsOverlapDiagram.stories.jsx:58]: are these printouts for tthe user?

File: src/Analysis/GWASV2/Shared/CohortsOverlapDiagram/CohortsOverlapDiagram.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
```
[the remark this answers is not in the document]
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/CohortsOverlapDiagram/CohortsOverlapDiagram.stories.jsx]: These console.log() statements have been removed now.

File: src/Analysis/GWASV2/Shared/CohortsOverlapDiagram/CohortsOverlapDiagram.stories.jsx (hunk 1 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/CohortsOverlapDiagram/CohortsOverlapDiagram.stories.jsx]: These console.log() statements have been removed now.

File: src/Analysis/GWASV2/Shared/AttritionTable/ChartIcons/BarChart.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        data-name='Path 4801'
+        d='M31.545,49.727H19.909A2.909,2.909,0,0,1,17,46.818V11.909A2.909,2.909,0,0,1,19.909,9H31.545a2.909,2.909,0,0,1,2.909,2.909V46.818A2.909,2.909,0,0,1,31.545,49.727Z'
+        transform='translate(30.545 15.273)'
+        fill='#5495d0'
+      />
+    </g>
+  </svg>
+);
+
+export default BarChart;
```
[the remark this answers is not in the document]
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTable/ChartIcons/BarChart.jsx:39]: Wasn't aware of that approach, I've created a ticket to refactor these icons and any other used in the new workflow to use that approach: https://ctds-planx.atlassian.net/browse/VADC-293

File: src/Analysis/GWASV2/Shared/AttritionTable/ChartIcons/EulerDiagram.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      id='Path_4817'
+      data-name='Path 4817'
+      d='M43.182,2.8A19.8,19.8,0,0,0,32.106,6.061,20.876,20.876,0,0,0,21.03,2.8,21.114,21.114,0,0,0,.106,24.13,21.114,21.114,0,0,0,21.03,45.464,19.8,19.8,0,0,0,32.106,42.2a20.876,20.876,0,0,0,11.076,3.264A21.114,21.114,0,0,0,64.106,24.13,21.114,21.114,0,0,0,43.182,2.8ZM32.106,10.7A17.008,17.008,0,0,1,34.2,12.834H30.015A17.29,17.29,0,0,1,32.106,10.7ZM27.059,30.4h9.97A16.051,16.051,0,0,1,35.8,32.913H28.169c-.37-.75-.74-1.632-1.11-2.509Zm-.736-10.037H37.768c.125.754.245,1.632.37,2.509H25.953C26.078,22,26.2,21.121,26.323,20.366Zm-.245,5.019H38.263a8.272,8.272,0,0,1-.37,2.509H26.444a8.273,8.273,0,0,1-.366-2.509Zm11.076-7.528h-9.97a16.051,16.051,0,0,1,1.231-2.509h7.629c.37.75.74,1.632,1.11,2.509ZM21.03,41.7a17.569,17.569,0,0,1,0-35.13,16.728,16.728,0,0,1,7.878,2.009,21.53,21.53,0,0,0,0,31.116,16.724,16.724,0,0,1-7.878,2Zm11.076-4.137a17.007,17.007,0,0,1-2.091-2.132H34.2a17.323,17.323,0,0,1-2.1,2.132Z'
+      transform='translate(-0.106 -2.797)'
+      fill='#2e77b8'
+    />
+  </svg>
+);
+
+export default EulerDiagram;
```
[the remark this answers is not in the document]
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Shared/AttritionTable/ChartIcons/EulerDiagram.jsx:22]: Wasn't aware of that approach, I've created a ticket to refactor these icons and any other used in the new workflow to use that approach: https://ctds-planx.atlassian.net/browse/VADC-293

**Review by alilahrime — APPROVED** [review]

**Pull request merged**